### PR TITLE
Updates to many panels to improve window scaling.

### PR DESCRIPTION
**Pull request by ghost**
Fixed GamePanel and JailPanel window scaling issues, fixed PieceCrafterMasterPanel scaling issues, minor changes to Buttons on PieceCrafterDetailPanel to remove broken button.

### CODE REVIEW

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/frame/GameFrame.java [hunk 1]
```diff
@@ -81,7 +81,12 @@ void onWindowCloseKeyboardRequest() {
 
     public void addGame(@NotNull Game game) {
         GamePanel panel = new GamePanel(getGlassPane(), game);
-        mTabbedPane.addTab(game.getGameType() + mTabbedPane.getTabCount(), panel);
+        if(mTabbedPane.getTabCount() != 0) {
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/frame/GameFrame.java]: We don't need this change, and even if we decide we want it, it needs to be in a separate commit

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java [hunk 2]
```diff
@@ -284,76 +259,60 @@ public void componentHidden(ComponentEvent e) {
...
-        constraints.fill = GridBagConstraints.HORIZONTAL;
-        constraints.anchor = GridBagConstraints.BASELINE;
-        constraints.gridwidth = 3;
-        constraints.gridheight = 1;
-        constraints.gridx = 11 + twoBoardsGridBagOffset;
-        constraints.gridy = 6;
-        constraints.insets = new Insets(0, 0, 0, 0);
+        constraints.gridy = 5;
+        // TODO: This assumes a two-player game
+//		jailAndLabelPanel.add(new ChessTimerLabel(mTimers[0]), constraints);
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java:301]: Please delete this if we don't need it

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java [hunk 3]
```diff
@@ -25,24 +25,10 @@ public JailPanel(int totalPieces) {
         mSquareLabels = new ArrayList<>(totalPieces);
 
         setBorder(GuiUtility.createBorder(Messages.getString("PlayGamePanel.capturedPieces")));
-        setPreferredSize(new Dimension((mJailDimension + 1) * 25, (mJailDimension + 1) * 25));
-
         setLayout(new GridLayout(mJailDimension, mJailDimension));
         createGrid();
     }
 
-    public void rescaleBoard(double panelWidth, double panelHeight) {
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java:34]: :+1: :+1: :+1:

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java [hunk 4]
```diff
@@ -468,7 +468,7 @@ private void initGUIComponents(PieceTypeBuilder builder) {
         constraints.gridwidth = 1;
         constraints.gridx = 0;
         constraints.gridy = 7;
-        add(cancelButton, constraints);
+        //add(cancelButton, constraints);
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java:471]: Remove this if we don't need it

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterMasterPanel.java [hunk 5]
```diff
@@ -41,16 +42,24 @@ private void initComponents() {
...
         scrollPane.setPreferredSize(new Dimension(190, 500));
 
-        add(scrollPane);
+        setLayout(new GridBagLayout());
+        GridBagConstraints constraints = new GridBagConstraints();
+
+        constraints.fill = GridBagConstraints.BOTH;
+        constraints.weightx = .5;
+        constraints.weighty = 1.0;
+        constraints.insets = new Insets(25,25,25,25);
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterMasterPanel.java]: `Ctrl+Alt+L` runs the code formatter on Windows, I believe. There should be spaces between the `25, 25, 25, 25`, so i think you need to run the formatter.

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java (hunk 2 above)
Comment (ghost) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java:301]: Can't delete it until we have a solution for Timers. Otherwise I'll forget where the go.

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java (hunk 4 above)
Comment (ghost) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java:471]: Not sure if it is needed or not, do we need people to be able to cancel their changes to a piece using a specific button?
